src/snowflake/snowpark/session.py

```diff
@@ -2008,6 +2014,7 @@ def create_dataframe(
                 quote_identifiers=True,
                 auto_create_table=True,
                 table_type="temporary",
+                use_logical_type=True,
```

Comment:
Is this a behavior change (i.e. wrong behavior -> correct behavior)? Should we highlight the implication in change log?

Reply:
I'll update the changelog detailing what exactly is going to change